ledger-core/virtual/object/object.go

```diff
@@ -332,42 +333,53 @@ func (sm *SMObject) stepWaitState(ctx smachine.ExecutionContext) smachine.StateU
 }
 
 func (sm *SMObject) stepGotState(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	state := sm.GetState()
+
 	if sm.PreviousExecutorOrderedPendingCount == 0 {
-		sm.releaseOrderedExecutionPath(ctx)
-		sm.releaseUnorderedExecutionPath(ctx)
-	} else if sm.GetState() != Empty {
-		sm.releaseUnorderedExecutionPath(ctx)
+		sm.releaseOrderedExecutionPath()
+		sm.releaseUnorderedExecutionPath()
+	} else if state != Empty {
```

Comment:
PreviousExecutorOrderedPendingCount type is uint8. It is always == 0 || > 0. Maybe we can combine this condition with next one?

Reply:
Well, we can't if I understand it right

Reply:
Oh, I've understand what you mean

Reply:
fixed